File: app/src/main/java/com/example/adouble/calibrationtest/Photo.java
package com.example.adouble.calibrationtest;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by Double on 2017/4/24.
 */

public class Photo {

    private static final String JSON_FILENAME = "path";

    private String path;

    public Photo(String name) {
        path = name;
    }

    public String getPath() {
        return path;
    }

    public Photo(JSONObject json) throws JSONException {
        path = json.getString(JSON_FILENAME);
    }

    public JSONObject toJSON() throws JSONException {
        JSONObject json = new JSONObject();
        json.put(JSON_FILENAME, path);
        return json;
    }
}

File: app/src/main/java/com/zjgsu/ai/calibrationtest/Photo.kt
package com.zjgsu.ai.calibrationtest

import org.json.JSONObject

/**
 * Created by Double on 28/07/2017.
 */
class Photo constructor(private val path: String) {
    companion object {
        private val JSON_FILENAME: String = "path"
    }

    constructor(json: JSONObject) : this(json.getString(JSON_FILENAME))

    fun getPath(): String { return path }

    fun toJSON(): JSONObject {
        var json: JSONObject = JSONObject()
        json.put(JSON_FILENAME, path)
        return json
    }
}